frappe/desk/reportview.py

```diff
@@ -54,35 +54,54 @@ def get_form_params():
 
 	fields = data["fields"]
 
+	if ((isinstance(fields, string_types) and fields == "*")
+		or (isinstance(fields, (list, tuple)) and len(fields) == 1 and fields[0] == "*")):
+		parenttype = data.doctype
+		data["fields"] = frappe.db.get_table_columns(parenttype)
+		fields = data["fields"]
+
 	for field in fields:
 		key = field.split(" as ")[0]
 
 		if key.startswith('count('): continue
 		if key.startswith('sum('): continue
 		if key.startswith('avg('): continue
 
-		if "." in key:
-			parenttype, fieldname = key.split(".")[0][4:-1], key.split(".")[1].strip("`")
-		else:
-			parenttype = data.doctype
-			fieldname = field.strip("`")
+		parenttype, fieldname = get_parent_dt_and_field(key, data)
 
-		df = frappe.get_meta(parenttype).get_field(fieldname)
+		if fieldname == "*":
+			# * inside list is not allowed with other fields
+			fields.remove(field)
+
+		meta = frappe.get_meta(parenttype)
+		df = meta.get_field(fieldname)
 
-		fieldname = df.fieldname if df else None
 		report_hide = df.report_hide if df else None
 
 		# remove the field from the query if the report hide flag is set and current view is Report
 		if report_hide and is_report:
 			fields.remove(field)
 
+		if df and fieldname in [df.fieldname for df in meta.get_high_permlevel_fields()]:
+			if df.get('permlevel') not in meta.get_permlevel_access() and field in fields:
```

Comment:
This fails for child table fields:
![image](https://user-images.githubusercontent.com/19775888/105277320-38892c00-5bc9-11eb-8e9a-eaa5563a071f.png)

(`parenttype` is not set in `get_permissions`)